lib/kube/grpc/grpc.go

```diff
@@ -264,41 +266,235 @@ func (s *Server) iterateKubernetesPods(
 	continueKey := req.StartKey
 	itemsAppended := 0
 	for {
-		podList, err := kubeClient.CoreV1().Pods(req.KubernetesNamespace).List(ctx, metav1.ListOptions{
-			Limit:    decideLimit(int64(req.Limit), int64(itemsAppended), respectLimit),
-			Continue: continueKey,
-		})
-		if err != nil {
-			return trace.Wrap(err)
+		var (
+			items           []kObject
+			nextContinueKey string
+			listOpts        = metav1.ListOptions{
+				Limit:    decideLimit(int64(req.Limit), int64(itemsAppended), respectLimit),
+				Continue: continueKey,
+			}
+		)
+
+		switch req.ResourceType {
+		case types.KindKubePod:
+			lItems, err := kubeClient.CoreV1().Pods(req.KubernetesNamespace).List(ctx, listOpts)
+			if err != nil {
+				return trace.Wrap(err)
+			}
+
+			items = itemListToKObjectList(itemListToItemListPtr(lItems.Items))
+			nextContinueKey = lItems.Continue
+
```

Comment:
nit: match spacing below
```suggestion
			items = itemListToKObjectList(itemListToItemListPtr(lItems.Items))
			nextContinueKey = lItems.Continue
```